pkg/test/e2e/expose-strategy/tunneling_test.go

```diff
@@ -20,73 +20,102 @@ package exposestrategy
-	)
-	ginkgo.BeforeEach(func() {
-		k8scli, restCli, restConf := e2eutils.GetClientsOrDie()
-		clusterJig = &ClusterJig{
-			Log:            logger,
-			Client:         k8scli,
-			Name:           "c" + rand.String(9),
-			DatacenterName: options.datacenter,
-			Version:        options.kubernetesVersion,
-		}
-		gomega.Expect(clusterJig.SetUp(ctx)).NotTo(gomega.HaveOccurred(), "user cluster should deploy successfully")
-		agentConfig = &AgentConfig{
-			Log:       logger,
-			Client:    k8scli,
-			Namespace: clusterJig.Cluster.Status.NamespaceName,
-			Versions:  kubermatic.NewDefaultVersions(),
-		}
-		client = &clientJig{e2eutils.TestPodConfig{
-			Log:           logger,
-			Namespace:     clusterJig.Cluster.Status.NamespaceName,
-			Client:        k8scli,
-			PodRestClient: restCli,
-			Config:        restConf,
-			CreatePodFunc: newClientPod,
-		}}
-		gomega.Expect(agentConfig.DeployAgentPod(ctx)).NotTo(gomega.HaveOccurred(), "agent should deploy successfully")
-		gomega.Expect(client.DeployTestPod(ctx, logger)).NotTo(gomega.HaveOccurred(), "client pod should deploy successfully")
-	})
-	ginkgo.AfterEach(func() {
-		if !options.skipCleanup {
-			gomega.Expect(clusterJig.CleanUp(ctx)).NotTo(gomega.HaveOccurred())
-			if agentConfig != nil {
-				gomega.Expect(agentConfig.CleanUp(ctx)).NotTo(gomega.HaveOccurred())
+	clusterJig := &ClusterJig{
+		Log:            logger,
+		Client:         k8scli,
+		Name:           "c" + rand.String(9),
+		DatacenterName: options.datacenter,
+		Version:        options.kubernetesVersion,
+	}
+	if err := clusterJig.SetUp(ctx); err != nil {
+		t.Errorf("Failed to setup usercluster: %v", err)
+		return
+	}
+
+	agentConfig := &AgentConfig{
+		Log:       logger,
+		Client:    k8scli,
+		Namespace: clusterJig.Cluster.Status.NamespaceName,
+		Versions:  kubermatic.NewDefaultVersions(),
+	}
+	if err := agentConfig.DeployAgentPod(ctx); err != nil {
+		t.Errorf("Failed to deploy agent: %v", err)
+		return
+	}
+
+	client := &clientJig{e2eutils.TestPodConfig{
+		Log:           logger,
+		Namespace:     clusterJig.Cluster.Status.NamespaceName,
+		Client:        k8scli,
+		PodRestClient: restCli,
+		Config:        restConf,
+		CreatePodFunc: newClientPod,
+	}}
+	if err := client.DeployTestPod(ctx, logger); err != nil {
+		t.Errorf("Failed to deploy Pod: %v", err)
+		return
+	}
+
+	logger.Debug("Testing SNI when Kubeconfig is used e.g. Kubelet")
+	if result := client.QueryApiserverVersion("", false, options.kubernetesVersion, 5, 4); result != true {
+		t.Error("Apiserver should be reachable passing from the SNI entrypoint in nodeport proxy")
+	}
+
+	logger.Debug("Tunneling requests using HTTP/2 CONNECT when no SNI is present e.g. pods relying on kubernetes service in default namespace")
+	if result := client.QueryApiserverVersion(agentConfig.GetKASHostPort(), true, options.kubernetesVersion, 5, 4); result != true {
+		t.Error("Apiserver should be reachable passing from the SNI entrypoint in nodeport proxy")
+	}
```

Comment:
This might be really nitpicky, but any chance we wrap those to into two `t.Run` calls (or make a simple table-driven test out of this)? That way, we will get separate test results and two entries in junit files (ref #10069), so we immediately see which test failed.

Reply:
Good idea!